File: leetcode/26/RemoveDuplicates.rb
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
    if (nums.length == 0 || nums.length == 1)
        return nums.length;
    end
    
    index = 0;
    1.step do |i| 
        if (nums[i-1] != nums[i])
            index = index +1;
            nums[index] = nums[i];
        end
        break if i>=nums.length-1 
    end
    return index + 1;
end

File: leetcode/26/RemoveDuplicates.ts
function removeDuplicates(nums: number[]): number {
    if (nums.length == 0 || nums.length == 1)
        return nums.length;
    let index = 0;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i - 1] != nums[i])
            nums[++index] = nums[i];
    }
    return index + 1;
};